### PR TITLE
Add attribute to overwrite the velocity constraints

### PR DESCRIPTION
**Pull request by tjstienstra**
#### Brief description of what is fixed or changed
This PR adds the `velocity_constraints` attribute to `System` to allow users to overwrite the velocity constraints, which can be computed automatically in most cases. An example where one is required to overwrite the velocity constraints is when computing the noncontributing forces in the closing joint of a four-bar linkage.

#### Other comments
The velocity constraints are only stored directly when specifically set by the user. Otherwise, they are computed based on the holonomic and nonholonomic constraints. This is to prevent possible inconsistencies in the system as the same constraint can/should be stored as both a nonholonomic constraint as well as a velocity constraint.

#### Release Notes

* physics.mechanics
  * Implemented System.velocity_constraints to allow overwriting the automatically computed velocity constraints.

### CODE REVIEW

File: doc/src/modules/physics/mechanics/examples/four_bar_linkage_example.rst [hunk 1]
```diff
@@ -80,3 +81,67 @@ using :class:`~.KanesMethod` as the backend. ::
    >>> simplify(system.form_eoms())
     Matrix([[l2*rho*(-2*l2**2*sin(q3)*u1' + 3*l2*l3*u1**2*sin(q2 + q3)*sin(q2) + 3*l2*l3*sin(q2)*cos(q2 + q3)*u1' - 3*l2*l3*sin(q3)*u1' + 3*l2*l4*u1**2*sin(q2 + q3)*sin(q2) + 3*l2*l4*sin(q2)*cos(q2 + q3)*u1' + 3*l3**2*u1**2*sin(q2)*sin(q3) + 6*l3**2*u1*u2*sin(q2)*sin(q3) + 3*l3**2*u2**2*sin(q2)*sin(q3) + 2*l3**2*sin(q2)*cos(q3)*u1' + 2*l3**2*sin(q2)*cos(q3)*u2' - l3**2*sin(q3)*cos(q2)*u1' - l3**2*sin(q3)*cos(q2)*u2' + 3*l3*l4*u1**2*sin(q2)*sin(q3) + 6*l3*l4*u1*u2*sin(q2)*sin(q3) + 3*l3*l4*u2**2*sin(q2)*sin(q3) + 3*l3*l4*sin(q2)*cos(q3)*u1' + 3*l3*l4*sin(q2)*cos(q3)*u2' + l4**2*sin(q2)*u1' + l4**2*sin(q2)*u2' + l4**2*sin(q2)*u3')/(6*sin(q3))]])
 
+Revealing noncontributing forces
+--------------------------------
+
+To reveal the noncontributing forces at the closing joint, we require to
```
Comment (moorepants) [inline doc/src/modules/physics/mechanics/examples/four_bar_linkage_example.rst:87]:
```suggestion
To reveal the noncontributing forces at the closing joint, we must
```

File: doc/src/modules/physics/mechanics/examples/four_bar_linkage_example.rst [hunk 2]
```diff
@@ -80,3 +81,67 @@ using :class:`~.KanesMethod` as the backend. ::
...
+   >>> end_point_forces = end_point.locatenew('end_point_forces', 0)
+   >>> end_point_forces.set_vel(N, uaux1*N.x + uaux2*N.y)
+   >>> noncontributing_forces = [Force(end_point_forces, faux1*N.x + faux2*N.y)]
+
+Next, we can add the auxiliary speeds and noncontributing forces to the system.
+
+   >>> system.add_loads(*noncontributing_forces)
+   >>> system.u_aux = [uaux1, uaux2]
+
+To include the gravity we can use :meth:`~.System.apply_uniform_gravity` before
```
Comment (moorepants) [inline doc/src/modules/physics/mechanics/examples/four_bar_linkage_example.rst:124]:
```suggestion
To include gravity we can use :meth:`~.System.apply_uniform_gravity` before
```

File: doc/src/modules/physics/mechanics/examples/four_bar_linkage_example.rst [hunk 3]
```diff
@@ -80,3 +81,67 @@ using :class:`~.KanesMethod` as the backend. ::
...
+   >>> subs = {
+   ...   l1: 2, l2: 1, l3: 2, l4: 1,
+   ...   rho: 5, g: 9.81,
+   ...   q1: pi/2, q2: pi/2, q3: pi/2,
+   ...   u1: 0, u2: 0, u3: 0, u1.diff(): 0, u2.diff(): 0, u3.diff(): 0,
+   ... }
+   >>> forces_eqs.xreplace(subs)
+   Matrix([
+   [    0],
+   [-98.1]])
```
Comment (moorepants) [inline doc/src/modules/physics/mechanics/examples/four_bar_linkage_example.rst:147]: Does that value look right? Magnitude an direction? It should be the proportion of weight on one end of how ever it is configured?

**Comment by moorepants** [comment]
This looks good to me and nice addition to the example. I'm not sure we have any example that shows how to do this for a closed loop any where. I left a couple small comments and maybe you can do a back of than calculation to see if that value is reasonable (if you haven't already).

File: doc/src/modules/physics/mechanics/examples/four_bar_linkage_example.rst (hunk 3 above)
Comment (tjstienstra) [reply doc/src/modules/physics/mechanics/examples/four_bar_linkage_example.rst:147]: The four-bar linkage is oriented as a perfect rectangle with link2 and link4 oriented upwards. Therefore, half of the mass of link3, `0.5 * 2 * 5 = 5`, and the full mass of link4, `1 * 5 = 5` rests on this joint. So a force of `9.81 * (5 + 5) = 98.1` acts downwards on the closing joint.

File: doc/src/modules/physics/mechanics/examples/four_bar_linkage_example.rst (hunk 2 above)
Comment (tjstienstra) [reply doc/src/modules/physics/mechanics/examples/four_bar_linkage_example.rst]: Addressed in 708248d96ca4879663334e43a2631a234c1bd997

File: doc/src/modules/physics/mechanics/examples/four_bar_linkage_example.rst (hunk 1 above)
Comment (tjstienstra) [reply doc/src/modules/physics/mechanics/examples/four_bar_linkage_example.rst]: Addressed in 708248d96ca4879663334e43a2631a234c1bd997

**Pull request merged**